scripts/run-functests.sh

```diff
@@ -1,17 +1,129 @@
 #!/bin/bash
+set -e -u -o pipefail
 
-# Show command before executing
-set -x
+declare -r CURRENT_DIR=$(pwd)
+declare -r SCRIPT_PATH=$(readlink -f "$0")
+declare -r SCRIPT_DIR=$(cd $(dirname "$SCRIPT_PATH") && pwd)
+declare -r PLANNER_PORT=8090
 
-echo "Building fabric8-planner..."
-npm install
+### Utility Functions
 
-echo "Run build fabric8-planner"
-mkdir -p dist && cp package.json dist/ && npm run build
+declare -r RED='\e[31m'
+declare -r GREEN='\e[32m'
+declare -r YELLOW='\e[33m'
+declare -r BLUE='\e[34m'
+declare -r MAGENTA='\e[35m'
+declare -r CYAN='\e[36m'
+declare -r WHITE='\e[37m'
+declare -r BOLD='\e[1m'
+declare -r RESET='\e[0m'
 
-echo "Building fabric8-planner/runtime..."
-cd runtime
-npm install
+_log() {
+  local status="$1"; shift
+  echo -e "${status}:$RESET ${*}$RESET" >&2
+}
 
-echo "Running smoke functional test..."
-npm run test:funcsmoke
+log.debug() {
+  debug_enabled || return 0
+  local caller_file=${BASH_SOURCE[1]##*/}
+  local caller_line=${BASH_LINENO[0]}
+  local caller_info="${WHITE}$caller_file${BLUE}(${caller_line}${BLUE})"
+  local caller_fn=""
+  if [ ${#FUNCNAME[@]} != 2 ]; then
+      caller_fn="${FUNCNAME[1]:+${FUNCNAME[1]}}"
+      caller_info+=" ${GREEN}$caller_fn"
+  fi
+  _log "${caller_info}" "$*" >&2
+}
+
+log.info()  { _log "$GREEN${BOLD}INFO" "$*"; }
+log.warn()  { _log "${YELLOW}WARNING" "$*"; }
+log.error() { _log "${RED}ERROR" "$*"; }
+
+debug_enabled() {
+  [[ ${DEBUG:-false} == true ]]
+}
+
+wait_for() {
+  # First arg is the command to execute
+  local WAIT_LIMIT=5
+  local NEXT_WAIT_TIME=0
+
+  until $1 || [ $NEXT_WAIT_TIME -eq $WAIT_LIMIT ]; do
+    sleep $(( NEXT_WAIT_TIME++ ))
+  done
+}
+
+## Process Control Functions
+
+check_port_free() {
+  # Check if port specified by $1 is free, else exit
+  if nc -z 127.0.0.1 $1; then
+    log.error "Cannot start planner. Port $PLANNER_PORT already in use by another process. Please free up the port and try again."
+    exit 1
+  fi
+}
+
+start_planner() {
+  check_port_free $PLANNER_PORT
+  log.info "NODE_ENV=inmemory mode set; Planner will use mock data"
+  NODE_ENV=inmemory npm run server:test -- --port $PLANNER_PORT &
```

Comment:
server:test emulates dev-server.
Can we rather emulate prod-server for the testing environment?

Reply:
Should make no difference and I don't want to do that much about the runtime, it will go away anyway.